interp/interp.go

```diff
@@ -337,6 +366,14 @@ func (interp *Interpreter) Repl(in, out *os.File) {
 	prompt := getPrompt(in, out)
 	prompt()
 	src := ""
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt)
+
+	go func() {
```

Comment:
This approach is one-shot. See https://github.com/kortschak/yaegi/commit/431f838ad4574f74d9ddba1245ab3de6b83e338e for an alternative.